client/my-sites/subscribers/components/subscribers-page/subscribers-page-context.tsx

```diff
@@ -37,16 +43,21 @@ const DEFAULT_PER_PAGE = 10;
 export const SubscribersPageProvider = ( {
 	children,
 	siteId,
-	page,
+	filterOption = SubscribersFilterBy.All,
+	pageNumber,
+	searchTerm,
+	sortTerm = SubscribersSortBy.DateSubscribed,
+	filterOptionChanged,
 	pageChanged,
+	searchTermChanged,
+	sortTermChanged,
 }: SubscribersPageProviderProps ) => {
-	const [ searchTerm, setSearchTerm ] = useState( '' );
 	const [ perPage, setPerPage ] = useState( DEFAULT_PER_PAGE );
-	const [ sortTerm, setSortTerm ] = useState( SubscribersSortBy.DateSubscribed );
-	const [ filterOption, setFilterOption ] = useState( SubscribersFilterBy.All );
 
 	const handleSearch = useCallback( ( term: string ) => {
-		setSearchTerm( term );
+		searchTermChanged( term );
+		pageChanged( 1 );
```

Comment:
Since Paulo will be AFK, I have addressed these changes in https://github.com/Automattic/wp-calypso/pull/79039/commits/e7862a386c485f1418190cf292eb6ff24ad08c6e.